rest/model/pod.go

```diff
@@ -0,0 +1,77 @@
+package model
+
+import (
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/evergreen-ci/utility"
+)
+
+// APISecretOpts is the model for secrets in a container.
+type APISecretOpts struct {
+	Name   *string
+	Value  *string
+	Owned  *bool
+	Exists *bool
+}
+
+// APIPodEnvVar is the model for environment variables in a container
+type APIPodEnvVar struct {
+	Name       *string
+	Value      *string
+	SecretOpts *APISecretOpts
+}
+
+// APITimeInfo is the model for the timing information of a pod.
+type APITimeInfo struct {
+	Initialized time.Time
+	Started     time.Time
+	Provisioned time.Time
+}
+
+// APIPod is the model to be returned by the API whenever pod.Pod is fetched.
+type APIPod struct {
+	Name     *string
+	Memory   *int
+	CPU      *int
+	Image    *string
+	EnvVars  []*APIPodEnvVar
+	Platform *string
+	Secret   *string
+}
+
+// fromAPIEnvVars converts a slice of APIPodEnvVars to a string-string map of environment variables and a string-string map of secrets.
+func fromAPIEnvVars(api []*APIPodEnvVar) (map[string]string, map[string]string) {
```

Comment:
Can you label the outputs as `(envVars map[string]string, secrets map[string]string)` to avoid any ambiguity in the order of the returned values? When two things are the same type and are next to each other in the input parameters or output types, it can be confusing to remember which comes first or second.